language/move-lang/src/parser/syntax.rs

```diff
@@ -307,6 +307,49 @@ fn parse_module_access<'input, F: FnOnce() -> String>(
     Ok(spanned(tokens.file_name(), start_loc, end_loc, acc))
 }
 
+//**************************************************************************************************
+// Visibility
+//**************************************************************************************************
+
+// Parse a function visibility modifier:
+//      FunctionVisibility = ( "public" ( "(" "script" ")" )? )?
+fn parse_visibility<'input>(tokens: &mut Lexer<'input>) -> Result<FunctionVisibility, Error> {
+    let visibility = if tokens.peek() == Tok::Public {
+        let start_loc = tokens.start_loc();
```

Comment:
I kind of prefer to keep the `start_loc` inside the condition because the `loc` is only related to public(*) visibilities and keeping it inside the branch aligns with the mentality. But I don't have a strong opinion on this.

Reply:
OK, there are precedents for doing it either way.